pb.c

```diff
@@ -1170,7 +1170,7 @@ typedef enum {USE_FIELD = 1, USE_REPEAT = 2, USE_MESSAGE = 4} lpb_DefFlags;
 static void lpb_pushtypetable(lua_State *L, lpb_State *LS, const pb_Type *t);
 
 static void lpb_newmsgtable(lua_State *L, const pb_Type *t)
-{ lua_createtable(L, 0, t->field_count - t->oneof_field + t->oneof_count*2); }
+{ lua_createtable(L, 0, t->field_count + t->oneof_count); }
```

Comment:
I cannot figure out how this lead the error, `t->oneof_count*2` is too big?

Reply:
Can the size of lua table be automatically scaled?

Reply:
The oneof_field can be bigger than t->field_count + t->oneof_count. According to the output of my gdb script, `t->field_count + t->oneof_count` is already enough to hold all the fields.